src/libstd/sync/mutex.rs

```diff
@@ -355,21 +339,14 @@ impl<T: ?Sized> Mutex<T> {
     pub fn get_mut(&mut self) -> LockResult<&mut T> {
         // We know statically that there are no other references to `self`, so
         // there's no need to lock the inner mutex.
-        let data = unsafe { &mut *self.data.get() };
+        let data = self.inner.get_mut();
         poison::map_result(self.poison.borrow(), |_| data )
     }
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
 unsafe impl<#[may_dangle] T: ?Sized> Drop for Mutex<T> {
-    fn drop(&mut self) {
-        // This is actually safe b/c we know that there is no further usage of
-        // this mutex (it's up to the user to arrange for a mutex to get
-        // dropped, that's not our job)
-        //
-        // IMPORTANT: This code must be kept in sync with `Mutex::into_inner`.
-        unsafe { self.inner.destroy() }
-    }
+    fn drop(&mut self) {}
```

Comment:
Why not remove the `Drop` impl entirely?

Reply:
Because that would change the public API of `Mutex`, would it not?

Reply:
Hm, fair, let's leave that for later. The behavior of `Drop` trait bounds might be changed.

Reply:
We could possibly try to remove it and run a crater run to verify not too many rely on `Drop` being implemented for these types.